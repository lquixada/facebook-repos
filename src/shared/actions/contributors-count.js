export const CONTRIBUTORS_COUNT_REQUESTED = 'CONTRIBUTORS_COUNT_REQUESTED';
export const CONTRIBUTORS_COUNT_SUCCEEDED = 'CONTRIBUTORS_COUNT_SUCCEEDED';
export const CONTRIBUTORS_COUNT_FAILED = 'CONTRIBUTORS_COUNT_FAILED';

export const fetchContributorsCount = () => ({
  type: CONTRIBUTORS_COUNT_REQUESTED,
});
