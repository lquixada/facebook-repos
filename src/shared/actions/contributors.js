export const CONTRIBUTORS_REQUESTED = 'CONTRIBUTORS_REQUESTED';
export const CONTRIBUTORS_SUCCEEDED = 'CONTRIBUTORS_SUCCEEDED';
export const CONTRIBUTORS_FAILED = 'CONTRIBUTORS_FAILED';

export const MORE_CONTRIBUTORS_REQUESTED = 'MORE_CONTRIBUTORS_REQUESTED';
export const MORE_CONTRIBUTORS_SUCCEEDED = 'MORE_CONTRIBUTORS_SUCCEEDED';
export const MORE_CONTRIBUTORS_FAILED = 'MORE_CONTRIBUTORS_FAILED';

export const fetchContributors = (repoName) => ({
  type: CONTRIBUTORS_REQUESTED,
  payload: {
    repoName
  }
});

export const fetchContributorsSucceeded = (repoName, data) => ({
  type: CONTRIBUTORS_SUCCEEDED,
  payload: {
    repoName,
    data
  }
});

export const fetchContributorsFailed = (repoName, error) => ({
  type: CONTRIBUTORS_FAILED,
  payload: {
    repoName,
    error
  }
});

export const fetchMoreContributors = (repoName, nextUrl) => ({
  type: MORE_CONTRIBUTORS_REQUESTED,
  payload: {
    repoName,
    nextUrl
  }
});

export const fetchMoreContributorsSucceeded = (repoName, data) => ({
  type: MORE_CONTRIBUTORS_SUCCEEDED,
  payload: {
    repoName,
    data
  }
});

export const fetchMoreContributorsFailed = (repoName, error) => ({
  type: MORE_CONTRIBUTORS_FAILED,
  payload: {
    repoName,
    error
  }
});
