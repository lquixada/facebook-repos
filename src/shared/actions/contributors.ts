export const CONTRIBUTORS_REQUESTED = 'CONTRIBUTORS_REQUESTED'
export const CONTRIBUTORS_SUCCEEDED = 'CONTRIBUTORS_SUCCEEDED'
export const CONTRIBUTORS_FAILED = 'CONTRIBUTORS_FAILED'

export const fetchContributors = (payload) => ({
  payload,
  type: CONTRIBUTORS_REQUESTED,
})

export const fetchContributorsSucceeded = (payload: {owner: string, repoName: string, data: any}) => ({
  payload,
  type: CONTRIBUTORS_SUCCEEDED,
})

export const fetchContributorsFailed = (payload: {owner: string, repoName: string, error: string}) => ({
  payload,
  type: CONTRIBUTORS_FAILED,
})
