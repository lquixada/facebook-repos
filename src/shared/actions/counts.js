export const COUNTS_REQUESTED = 'COUNTS_REQUESTED'
export const COUNTS_SUCCEEDED = 'COUNTS_SUCCEEDED'
export const COUNTS_FAILED = 'COUNTS_FAILED'

export const fetchCounts = (owner) => ({
  type: COUNTS_REQUESTED,
  payload: {
    owner
  }
})

export const fetchCountsSucceeded = (payload) => ({
  type: COUNTS_SUCCEEDED,
  payload
})

export const fetchCountsFailed = (error) => ({
  type: COUNTS_FAILED,
  payload: {
    error
  }
})
