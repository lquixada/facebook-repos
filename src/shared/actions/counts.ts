export const COUNTS_REQUESTED = 'COUNTS_REQUESTED'
export const COUNTS_SUCCEEDED = 'COUNTS_SUCCEEDED'
export const COUNTS_FAILED = 'COUNTS_FAILED'

export const fetchCounts = (owner) => ({
  payload: {
    owner,
  },
  type: COUNTS_REQUESTED,
})

export const fetchCountsSucceeded = (payload) => ({
  payload,
  type: COUNTS_SUCCEEDED,
})

export const fetchCountsFailed = (owner, error) => ({
  payload: {
    error,
    owner,
  },
  type: COUNTS_FAILED,
})
