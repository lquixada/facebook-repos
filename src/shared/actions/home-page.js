export const HOME_PAGE_REQUESTED = 'HOME_PAGE_REQUESTED'
export const HOME_PAGE_SUCCEEDED = 'HOME_PAGE_SUCCEEDED'
export const HOME_PAGE_FAILED = 'HOME_PAGE_FAILED'

export const loadHomePage = () => ({
  type: HOME_PAGE_REQUESTED
})
