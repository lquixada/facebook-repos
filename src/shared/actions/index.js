export * from './contributors'
export * from './home-page'
export * from './repo'
export * from './repos'
export * from './repo-page'
