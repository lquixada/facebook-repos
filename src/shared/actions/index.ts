export * from './contributors'
export * from './counts'
export * from './pages'
export * from './repo'
