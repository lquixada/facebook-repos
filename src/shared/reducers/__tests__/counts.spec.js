import {fetchCounts, fetchCountsSucceeded, fetchCountsFailed} from '../../actions'
import reducer from '../counts'

describe('Reducers (Counts)', () => {
  let owner
  let repoName

  beforeEach(() => {
    owner = 'owner1'
    repoName = 'repo1'
  })

  it('starts empty', () => {
    const state = reducer()
    expect(state.toJS()).toEqual({})
  })

  it('switches to isLoading state', () => {
    const action = fetchCounts(owner)
    const state = reducer(undefined, action)

    expect(state.getIn([owner, 'isLoading'])).toBe(true)
    expect(state.getIn([owner, 'error'])).toBe(null)
  })

  it('sets the counts state', () => {
    const data = [{name: repoName, count: 2}, {name: 'repo2', count: 5}]
    const action = fetchCountsSucceeded({owner, data})
    const state = reducer(undefined, action)

    expect(state.toJS()).toMatchSnapshot()
  })

  it('sets an error state', () => {
    const action = fetchCountsFailed(owner, 'some-error')
    const state = reducer(undefined, action)

    expect(state.getIn([owner, 'isLoading'])).toBe(false)
    expect(state.getIn([owner, 'error'])).toBe('some-error')
  })
})
