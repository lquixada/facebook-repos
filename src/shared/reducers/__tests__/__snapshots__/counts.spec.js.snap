// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers (Counts) sets the counts state 1`] = `
Object {
  "owner1": Object {
    "data": Array [
      Array [
        "repo2",
        5,
      ],
      Array [
        "repo1",
        2,
      ],
    ],
    "error": null,
    "isLoading": false,
  },
}
`;
